src/sidebar/components/Annotation/AnnotationHeader.js

```diff
@@ -113,10 +105,21 @@ function AnnotationHeader({
     // an ID.
     store.setExpanded(/** @type {string} */ (annotation.id), true);
 
+  const group = store.getGroup(annotation.group);
+
+  /** @param {{ children: import("preact").ComponentChildren}} props */
+  function HeaderRow({ children }) {
```

Comment:
This should be moved to the top level of the file. Declaring it inline within this component will change its identity on each render (of this component), which will cause Preact to do more work than necessary, since it will think the component type changed, rather than being re-rendered with different props.